gpiozero/spi_devices.py

```diff
@@ -351,23 +455,33 @@ class MCP3208(MCP32xx):
     """
     def __init__(self, channel=0, differential=False, **spi_args):
         if not 0 <= channel < 8:
-            raise InputDeviceError('channel must be between 0 and 7')
+            raise SPIBadChannel('channel must be between 0 and 7')
         super(MCP3208, self).__init__(channel, differential, **spi_args)
 
 
 class MCP3301(MCP33xx):
     """
     The `MCP3301`_ is a signed 13-bit analog to digital converter.  Please note
-    that the MCP3301 always operates in differential mode between its two
-    channels and the output value is scaled from -1 to +1.
+    that the MCP3301 always operates in differential mode measuring the
+    difference between IN+ and IN-. Its output value is scaled from -1 to +1.
 
     .. _MCP3301: http://www.farnell.com/datasheets/1669397.pdf
     """
     def __init__(self, **spi_args):
         super(MCP3301, self).__init__(0, differential=True, **spi_args)
 
-    def _send(self):
-        return [0, 0]
+    def _read(self):
+        # MCP3301 protocol looks like the following:
+        #
+        #     Byte        0        1
+        #     ==== ======== ========
+        #     Rx   xx0SRRRR RRRRRRRR
+        result = self._words_to_int(self._spi.read(2), 13)
+        # Account for the sign bit
+        if result > 4095:
+            return -(8192 - result)
+        else:
+            return result
```

Comment:
In the spirit of DRY, I wonder if this `result > 4095` snippet could be shared in a common function with the base `MCP33xx._read` method?